notebooks/search/09-semantic-text.ipynb

```diff
@@ -0,0 +1,510 @@
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "ad417b4b3f50c889"
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Semantic Search with the `semantic` Query\n",
+    "\n",
+    "We can use the `semantic` query to quickly & easily query the `semantic_text` field in our index.\n",
+    "Under the hood, an embedding is automatically generated for our query text using the `semantic_text` field's inference endpoint.\n",
+    "\n",
+    "TODO: Add link to `semantic` query documentation"
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "22c4d4d395adb472"
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "response = client.search(\n",
+    "    index=\"semantic-text-movies\",\n",
+    "    query={\"semantic\": {\"field\": \"plot_semantic\", \"query\": \"organized crime movies\"}},\n",
+    ")\n",
+    "\n",
+    "pretty_search_response(response)"
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "1a8520ffc8a3efb3"
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "These results demonstrate the power of semantic search.\n",
+    "Our top results are all movies involving organized crime, even if the exact term \"organized crime\" doesn't appear in the plot description.\n",
+    "This works because the ELSER model understands the semantic similarity between terms like \"organized crime\" and \"mob\".\n",
+    "\n",
+    "However, these results also show the weaknesses of semantic search.\n",
+    "Because semantic search is based on vector similarity, there is a long tail of results that are weakly related to our query vector.\n",
+    "That's why movies like _The Matrix_ are returned towards the tail end of our search results."
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "148fda24a3964aa9"
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Hybrid Search with the `semantic` Query\n",
+    "\n",
+    "We can address some of the issues with pure semantic search by combining it with lexical search techniques.\n",
+    "Here, we use a [boolean query](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-bool-query.html) to require that all matches contain at least term from the query text, in either the `plot` or `genre` fields."
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "7c9bab225a745746"
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "response = client.search(\n",
+    "    index=\"semantic-text-movies\",\n",
+    "    query={\n",
+    "        \"bool\": {\n",
+    "            \"must\": {\n",
+    "                \"multi_match\": {\n",
+    "                    \"fields\": [\"plot\", \"genre\"],\n",
+    "                    \"query\": \"organized crime movies\",\n",
+    "                }\n",
+    "            },\n",
+    "            \"should\": {\n",
```

Comment:
Should we include boosting to let users know about linear combination of results?

Reply:
Hmm, how far should we take that? Tuning linear combination is complex enough to deserve its own notebook. Adding a `boost` param to the example is simple, but doesn't do much to explain what we're doing and why. And on the other side of the spectrum, adding a whole section about linear combination steals focus from the main topic of this notebook.

Reply:
Alternatively, we could turn the `multi_match` query into a filter to side-step all that potential complexity in this example.

Reply:
My concern is that users may not get good results for hybrid search. It might be good to show them the way for doing linear combination by exposing that particular knob to turn.

Agreed that this deserves its own blog post or link to documentation - just wanted to raise awareness for users that this is something they will need for hybrid search.

Reply:
Would it be sufficient to add some example boosts and a short blurb about why those are there?

Reply:
For me, yes - nothing fancy, just letting users understand that need to be aware of linear combination for hybrid search 👍

Reply:
Added, LMKWYT!

Reply:
Looks good, thanks! 💯